### PR TITLE
[Component with its own class name as service type] LPS-194841 Refactor Interpreter component 

### PR DESCRIPTION
**Pull request by Valmir-unicap**

### CODE REVIEW

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:sf

File: modules/apps/sharing/sharing-web/src/main/java/com/liferay/sharing/web/internal/interpreter/AssetRendererSharingEntryInterpreter.java [hunk 1]
```diff
@@ -25,17 +25,33 @@
...
+		ServletContext servletContext) {
+
+		_assetEntryLocalService = assetEntryLocalService;
+		_assetRendererSharingEntryEditRenderer =
+			assetRendererSharingEntryEditRenderer;
+		_assetRendererSharingEntryViewRenderer =
+			assetRendererSharingEntryViewRenderer;
+
+		assetRendererSharingEntryEditRenderer =
+			new AssetRendererSharingEntryEditRenderer();
```
Comment (rafaprax) [inline modules/apps/sharing/sharing-web/src/main/java/com/liferay/sharing/web/internal/interpreter/AssetRendererSharingEntryInterpreter.java:49]: @Valmir-unicap, you do not need to do those "reinitializations". Please remove.

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/apps/sharing/sharing-web/src/main/java/com/liferay/sharing/web/internal/interpreter/SharingEntryInterpreterProviderImpl.java [hunk 2]
```diff
@@ -70,7 +70,6 @@ private boolean _isAssetObject(String className) {
 		return false;
 	}
 
-	@Reference
 	private AssetRendererSharingEntryInterpreter
```
Comment (rafaprax) [inline modules/apps/sharing/sharing-web/src/main/java/com/liferay/sharing/web/internal/interpreter/SharingEntryInterpreterProviderImpl.java:73]: Please initialize this field.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by Valmir-unicap** [comment]
see https://github.com/Valmir-unicap/liferay-portal/pull/44

**Pull request closed without merging**